File: app/service.rb
class Service
  attr_reader :name, :id, :type, :color

  def initialize
    @name = ''
    @id = ''
    @type = ''
    @color = ''
  end

  def name_append(string)
    @name << string
  end

  def id_append(string)
    @id << string

    @id = Colors.routes[@id] if Colors.routes.has_key?(@id)
  end

  def type_append(string)
    @type << string
  end

  def color_append(string)
    @color << string
  end

  def uicolor
    return UIColor.blackColor if @color.length < 6

    red = @color[0..1].to_i(16).to_f / 255.0
    green = @color[2..3].to_i(16).to_f / 255.0
    blue = @color[4..5].to_i(16).to_f / 255.0

    UIColor.colorWithRed(red, green: green, blue: blue, alpha: 1.0)
  end

  def passes_filter?(filter_list)
    filter_list.index(@id) != nil
  end
end

File: Rail Line/Service.swift
import UIKit

class Service {
    var name = ""
    private var type = ""
    private var color = ""

    func nameAppend(_ string: String) {
        name.append(contentsOf: string)
    }

    func typeAppend(_ string: String) {
        type.append(contentsOf: string)
    }

    func colorAppend(_ string: String) {
        color.append(contentsOf: string)
    }

    func typeIsTrain() -> Bool {
        return type.contains("Train")
    }

    func getColor() -> UIColor {
        if color.count != 6 { return Colors.black }

        var rgbValue: UInt32 = 0
        Scanner(string: color).scanHexInt32(&rgbValue)

        let red = CGFloat((rgbValue & 0xFF0000) >> 16)
        let green = CGFloat((rgbValue & 0x00FF00) >> 8)
        let blue = CGFloat(rgbValue & 0x0000FF)

        return UIColor(red: red / 255.0, green: green / 255.0, blue: blue / 255.0, alpha: 1.0)
    }
}
